lib/models/entitiesStore.js

```diff
@@ -28,15 +28,89 @@ var async = require('async'),
     entitiesCollectionName = require('../../config').orionDb.collection,
     myutils = require('../myutils'),
     constants = require('../constants'),
-    logger = require('logops');
+    logger = require('logops'),
+    NGSI = require('ngsijs');
 
 function orionServiceDb(service) {
     return appContext.OrionDb(config.orionDb.prefix + '-' + service);
 }
 
-function findSilentEntities(service, subservice, ruleData, func, callback) {
+function findSilentEntitiesByAPI(service, subservice, ruleData, func, callback) {
+    var context = { op: 'checkNoSignal.findSilentEntitiesByAPI', comp: constants.COMPONENT_NAME };
+
+    var options = {
+        service: service,
+        servicepath: subservice
+    };
+    options.headers = {};
+    // if (token !== null) {
+    //     options.headers[constants.AUTH_HEADER] = token;
+    // }
+    // Add correlator
+    var domain = process.domain;
+    if (domain && domain.context) {
+        options.headers[constants.CORRELATOR_HEADER] = domain.context.corr;
+        // Add other headers
+        if (domain.context.srv && options.headers[constants.SERVICE_HEADER] === undefined) {
+            options.headers[constants.SERVICE_HEADER] = domain.context.srv;
+        }
+        if (domain.context.subsrv && options.headers[constants.SUBSERVICE_HEADER] === undefined) {
+            options.headers[constants.SUBSERVICE_HEADER] = domain.context.subsrv;
+        }
+        if (domain.context.from && options.headers[constants.REALIP_HEADER] === undefined) {
+            options.headers[constants.REALIP_HEADER] = domain.context.from;
+        }
+    }
+    var connection = new NGSI.Connection(config.orion.URL, options);
+
+    var filter = {
+        service: service,
+        servicepath: subservice,
+        type: ruleData.type,
+        mq: ruleData.attribute + '.dateModified<' + (Date.now() / 1000 - ruleData.reportInterval).toString()
+    };
+    if (ruleData.id) {
+        filter.id = ruleData.id;
+    } else if (ruleData.idRegexp) {
+        filter.idPattern = ruleData.idRegexp;
+    }
+    logger.debug(
+        context,
+        'find silent entities by API ngsi using options %j and filter %j and rule %j',
+        options,
+        filter,
+        ruleData
+    );
+    // https://ficodes.github.io/ngsijs/stable/NGSI.Connection.html#.%22v2.listEntities%22__anchor
+    connection.v2.listEntities(filter).then(
+        (response) => {
+            // Entities retrieved successfully
+            // response.correlator transaction id associated with the server response
+            // response.limit contains the used page size
+            // response.results is an array with the retrieved entities
+            // response.offset contains the offset used in the request
+            var count = 0;
+            response.results.forEach((entity) => {
+                logger.debug(context, 'silent entity %j', entity);
+                func(entity);
```

Comment:
func has no meaning. Use some meaningful identifier for "func" param

Reply:
Solved in this commit: https://github.com/telefonicaid/perseo-fe/pull/726/commits/a0b86c35ecacfe6d11f6b260fdc33666bb5fe214